### PR TITLE
 create file structure for server side and install almost packeg we w…

### LINKED ISSUE #1: [ File structure ] src folder server side
src
     | - __tests__
     | - config
     | - controllers
     | - db
     | - helpers
     | - interfaces
     | - middlewares
     | - models
     | - queries
     | - routes
     | - validation
     | - app.ts
     | - index.ts
         package.json
         .eslintrc
         tsconfig.json

install:
  - express
  - jest
  - nodewatch
  - sokite
  - compression
  - bcryptjs
  - jsonwebtoken
  - joi
  - pg
  - cookies-parser
  - eslint
  - dotenv
  - prettier
  - HTTP-error
  - supetest

### PR DESCRIPTION
**Pull request by husamkamal**
…ant relates #1

### CODE REVIEW

File: src/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "typescript": "^4.8.4"
+  },
+  "dependencies": {
+    "bcryptjs": "^2.4.3",
+    "compression": "^1.7.4",
+    "cookies-parser": "^1.2.0",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "http-error": "^0.0.6",
+    "jest": "^29.1.2",
```
Comment (LinaYahya) [inline src/package.json]:
```suggestion
```
install as a devdependancy

File: src/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "eslint": "^8.25.0",
+    "typescript": "^4.8.4"
+  },
+  "dependencies": {
+    "bcryptjs": "^2.4.3",
+    "compression": "^1.7.4",
+    "cookies-parser": "^1.2.0",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "http-error": "^0.0.6",
```
Comment (LinaYahya) [inline src/package.json:35]: Are we going to use it?

File: src/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "jest": "^29.1.2",
+    "joi": "^17.6.2",
+    "jsonwebtoken": "^8.5.1",
+    "nodemon": "^2.0.20",
+    "pg": "^8.8.0",
+    "prettier": "^2.7.1",
```
Comment (LinaYahya) [inline src/package.json]:
```suggestion
```

File: src/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "sequelize": "^6.24.0",
+    "socket.io": "^4.5.2",
+    "supertest": "^6.3.0"
```
Comment (LinaYahya) [inline src/package.json]:
```suggestion
```

File: src/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.39.0",
```
Comment (LinaYahya) [inline src/package.json]: Why `airbnb` not used as a style guide?

File: src/package.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "src",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (LinaYahya) [inline src/package.json]: add a script dev to run index.ts using nodemon,
have a look at [ts-node](https://www.npmjs.com/package/ts-node)

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by mai-mohammed — APPROVED** [review]

**Review by tariqabusada — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**